src/server/session.ts

```diff
@@ -1765,6 +1766,11 @@ namespace ts.server {
                 this.setCompilerOptionsForInferredProjects(request.arguments);
                 return this.requiredResponse(true);
             },
+            [CommandNames.LoadTypesMap]: (request: protocol.FileRequest) => {
+                const loadArgs = <protocol.FileRequestArgs>request.arguments;

```

Comment:
```FileRequestArgs``` has a ```projectFileName``` member which you never look at.  While it's optional, it still gives the false impression that project context matters for this message (it doesn't -- it loads the same type map for all projects). I'd use a different type that only contains ```file```.